web/src/components/Header/HeaderMenu.tsx

```diff
@@ -0,0 +1,70 @@
+import {QuestionCircleOutlined} from '@ant-design/icons';
+import {Popover} from 'antd';
+import {Dispatch, SetStateAction, useMemo} from 'react';
+import {useParams} from 'react-router-dom';
+import {DOCUMENTATION_URL, GITHUB_URL} from '../../constants/Common.constants';
+import * as S from './Header.styled';
+import {ShowOnboardingContent} from './ShowOnboardingContent';
+
+interface IProps {
+  visible: boolean;
+  pathname: string;
+  setIsOpen: Dispatch<SetStateAction<Boolean>>;
+  setVisible: Dispatch<SetStateAction<boolean>>;
+  onGuidedTourClick: () => void;
+  title: () => JSX.Element;
+}
+
+export const HeaderMenu = ({pathname, setIsOpen, title, visible, setVisible, onGuidedTourClick}: IProps) => {
+  const handleGuidedTourCLick = () => setVisible(o => !o);
+  const content = useMemo(
```

Comment:
I don't think this useMemo is adding value here.

Reply:
I am pretty sure it avoid calling the content function on every render